src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -357,6 +358,21 @@ private static unsafe bool TryParseNumber(ref char* str, char* strEnd, NumberSty
                         {
                             number.Scale++;
                         }
+                        else
+                        {
+                            if (digCount < maxDigCount)
```

Comment:
nit: this could just be `else if` since there is no other logic in the `else` block